api/v1alpha1/httproute_types.go

```diff
@@ -264,6 +264,59 @@ type ForwardToTarget struct {
 	//
 	// +optional
 	TargetPort *TargetPort `json:"targetPort" protobuf:"bytes,2,opt,name=targetPort"`
+
+	// TimeoutPolicy describes a policy for configuring connection timeouts to targetRef.
+	//
+	// Support: Core
+	//
+	// +optional
+	TimeoutPolicy *TimeoutPolicy `json:"timeoutPolicy,omitempty" protobuf:"bytes,3,opt,name=timeoutPolicy"`
+
+	// RetryPolicy describes a policy for retrying a connection to a targetRef.
+	//
+	// Support: Core
+	//
+	// +optional
+	RetryPolicy *RetryPolicy `json:"retryPolicy,omitempty" protobuf:"bytes,4,opt,name=retryPolicy"`
+}
+
+// TimeoutPolicy defines a schema for configuring connection timeouts.
+// The format is: h/m/s/ms with 1ms being the smallest interval.
+type TimeoutPolicy struct {
+	// Response is the time allowed for receiving a response after a
+	// request has been forwarded.
+	//
+	// Support: Core
+	//
+	Response metav1.Duration `json:"response" protobuf:"bytes,1,opt,name=response"`
```

Comment:
Is this an inactivity timeout or a deadline to complete the transfer?  The field's description is suggests it is a deadline (for example, if the response is large, and it takes too long, a proxy may terminate the transfer even while data are being transferred), in which case it should be named as follows:
```suggestion
	ResponseDeadlineSeconds metav1.Duration `json:"response" protobuf:"bytes,1,opt,name=responseDeadlineSeconds"`
```
On the other hand, if this is an inactivity timeout (for example, if the server doesn't send any data within this amount of time, a proxy may terminate the connection), then it should be named as follows:
```suggestion
	ResponseTimeoutSeconds metav1.Duration `json:"response" protobuf:"bytes,1,opt,name=responseTimeoutSeconds"`
```
See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions.

Reply:
On second thought, maybe "Seconds" should be omitted for `metav1.Duration`, which is unmarshalled using `time.ParseDuration` and thus accepts time-unit suffixes (https://github.com/kubernetes/apimachinery/blob/2456ebdaba229616fab2161a615148884b46644b/pkg/apis/meta/v1/duration.go#L31-L45).  Oddly, I cannot find any APIs in k8s.io/api that use `metav1.Duration`.  Do you have examples of Kubernetes APIs that do use `metav1.Duration`?

Reply:
Ah, https://github.com/kubernetes/cloud-provider/blob/master/config/types.go has some examples of fields that use `metav1.Duration`.

So I think the field should be named either `responseDeadline` or `responseTimeout`.